internal/apiclient/pricing.go

```diff
@@ -169,26 +270,123 @@ func (c *PricingAPIClient) BatchRequests(resources []*schema.Resource, batchSize
 	return reqs
 }
 
+type pricingQuery struct {
+	hash  uint64
+	query GraphQLQuery
+
+	result gjson.Result
+}
+
+// PerformRequest sends a batch request to the Pricing API endpoint to fetch
+// pricing details for the provided queries. It optimizes the API call by
+// checking a local cache for previous results. If the results of a given query
+// are cached, they are used directly; otherwise, a request to the API is made.
 func (c *PricingAPIClient) PerformRequest(req BatchRequest) ([]PriceQueryResult, error) {
 	log.Debugf("Getting pricing details for %d cost components from %s", len(req.queries), c.endpoint)
+	res := make([]PriceQueryResult, len(req.keys))
+	for i, key := range req.keys {
+		res[i].PriceQueryKey = key
+	}
+
+	queries := make([]pricingQuery, len(req.queries))
+	for i, query := range req.queries {
+		key, err := hashstructure.Hash(query, hashstructure.FormatV2, nil)
+		if err != nil {
+			logging.Logger.WithError(err).Debugf("failed to hash query %s will use nil hash", query)
+		}
+
+		queries[i] = pricingQuery{
+			hash:  key,
+			query: query,
+		}
+	}
 
-	results, err := c.doQueries(req.queries)
+	// first filter any queries that have been stored in the cache. We don't need to
+	// send requests for these as we already have the results in memory.
+	var serverQueries []pricingQuery
+	if c.cache != nil {
+		for i, query := range queries {
+			c.mu.RLock()
+			v, ok := (*c.cache)[query.hash]
+			c.mu.RUnlock()
+			if ok {
+				res[i] = PriceQueryResult{
+					PriceQueryKey: req.keys[i],
+					Result:        v,
+					filled:        true,
+				}
+			} else {
+				serverQueries = append(serverQueries, query)
+			}
```

Comment:
Should we debug log if it's a cache hit or not, so we can look through the logs to see what % hit rate we are getting?